compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.h

```diff
@@ -111,6 +111,9 @@ SmallVector<Value> getTileSizes(OpBuilder &b, Operation *op, unsigned level);
 /// Sets the lowering configuration, overwriting existing attribute values.
 void setLoweringConfig(Operation *op, Attribute config);
 
+/// Sets the root operation mark in attributes.
+void setRootOpInfo(Operation *op, StringAttr config);
```

Comment:
Maybe something like:

```suggestion
/// Adds an attribute (`root_op_info`) to `op` to mark it as the root op.
void setRootOpInfo(Operation *op, Attribute info);
```